packages/core/test/polygon/e2e.js

```diff
@@ -0,0 +1,176 @@
+
+const { utf8ToHex, toWei, hexToUtf8 } = web3.utils;
+
+const StateSync = artifacts.require("StateSyncMock");
+const FxChild = artifacts.require("FxChildMock");
+const FxRoot = artifacts.require("FxRootMock");
+const OracleChildTunnel = artifacts.require("OracleChildTunnel");
+const OracleRootTunnel = artifacts.require("OracleRootTunnelMock");
+const Finder = artifacts.require("Finder");
+const Registry = artifacts.require("Registry");
+const MockOracle = artifacts.require("MockOracleAncillary");
+const IdentifierWhitelist = artifacts.require("IdentifierWhitelist");
+
+contract("Polygon <> Ethereum Tunnel: End-to-End Test", async (accounts) => {
+  const owner = accounts[0];
+  const systemSuperUser = accounts[1];
+  const checkpointManager = accounts[2];
+  const rando = accounts[3];
+
+  let stateSync;
+  let fxChild;
+  let fxRoot;
+  let oracleChild;
+  let oracleRoot;
+
+  // Oracle system:
+  let finder;
+  let identifierWhitelist;
+  let registry;
+  let mockOracle;
+
+  const testIdentifier = utf8ToHex("TEST");
+  const testTimestamp = 100;
+  const testAncillaryData = utf8ToHex("key:value");
+  const testPrice = toWei("0.5");
+  let expectedStampedAncillaryData; // Can determine this after contracts are deployed.
+  const expectedStateId = "1";
+
+  before(async function () {
+    finder = await Finder.deployed();
+    identifierWhitelist = await IdentifierWhitelist.deployed();
+    registry = await Registry.deployed();
+    mockOracle = await MockOracle.new(finder.address, ZERO_ADDRESS);
+
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.Registry), registry.address);
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.Oracle), mockOracle.address);
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.IdentifierWhitelist), identifierWhitelist.address);
+
+    await registry.addMember(RegistryRolesEnum.CONTRACT_CREATOR, owner);
+    await registry.registerContract([], owner, { from: owner });
+
+    await identifierWhitelist.addSupportedIdentifier(testIdentifier, { from: owner });
+  });
+  beforeEach(async function () {
+    // Set up mocked Fx tunnel system:
+    stateSync = await StateSync.new();
+    fxRoot = await FxRoot.new(stateSync.address);
+    fxChild = await FxChild.new(systemSuperUser);
+    await fxChild.setFxRoot(fxRoot.address);
+    await fxRoot.setFxChild(fxChild.address);
+
+    // Set up Oracle tunnel system:
+    oracleChild = await OracleChildTunnel.new(fxChild.address, finder.address);
+    oracleRoot = await OracleRootTunnel.new(checkpointManager, fxRoot.address, finder.address);
+    await oracleChild.setFxRootTunnel(oracleRoot.address);
+    await oracleRoot.setFxChildTunnel(oracleChild.address);
+    await registry.registerContract([], oracleRoot.address, { from: owner });
+
+    expectedStampedAncillaryData = utf8ToHex(
+      `${hexToUtf8(testAncillaryData)},childRequester:${owner.substr(2).toLowerCase()},childChainId:31337`
```

Comment:
prehaps put childChain as a variable?

Also, it would be nice for you to show in a comment the exact expected form. you _can_ re-construct it but there is a fair amount of nested calls here so a comment might help a little bit.